.github/workflows/BuildAndDeploy.yaml

```diff
@@ -42,7 +42,7 @@ on:
         default: 11
         type: string
       generate_gcp_general_secret:
-        default: "false"
+        default: ""
```

Comment:
How is the default "" value will be used if user did not provide false/true?

Reply:
We have described it in [documentation](https://cariq.atlassian.net/wiki/spaces/INFRA/pages/1780777011/Onboard+Your+Service+To+CI+CD+v2#Workflow-Parameters-(Customization)).
Find the description for the 14th parameter. We are faced with the fact that different services use different secrets  and it was decided do not add additional parameters for the secret name.
we used condition for creation of cred file.
` - name: Save credentials for build steps for some images
        if: ${{ inputs.generate_gcp_general_secret != '' }}
        shell: bash
        run: |
          gcloud secrets versions access latest \
          --secret=${{ inputs.generate_gcp_general_secret }} \
          --format='get(payload.data)' | tr '_-' '/+' | base64 -d > \
          gcp-general.json`